### PR TITLE
Convert PublicSuffixDatabase to Kotlin

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt [hunk 1]
```diff
@@ -13,131 +13,109 @@
...
-      firstLabelOffset = domainLabels.length - (rule.length + 1);
+      domainLabels.size - (rule.size + 1)
     }
 
-    StringBuilder effectiveTldPlusOne = new StringBuilder();
-    String[] punycodeLabels = domain.split("\\.");
-    for (int i = firstLabelOffset; i < punycodeLabels.length; i++) {
-      effectiveTldPlusOne.append(punycodeLabels[i]).append('.');
+    return buildString {
+      val punycodeLabels = domain.split("\\.".toRegex())
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt]: domain.split('.')

File: okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt [hunk 2]
```diff
@@ -13,131 +13,109 @@
...
+      for (i in firstLabelOffset until punycodeLabels.size) {
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt]: punycodeLabels.asSequence().drop(firstLabelOffset).joinToString(".")?

File: okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt [hunk 3]
```diff
@@ -146,195 +124,212 @@ public final class PublicSuffixDatabase {
...
         }
       }
     }
 
     if (exception != null) {
       // Signal we've identified an exception rule.
-      exception = "!" + exception;
-      return exception.split("\\.");
+      exception = "!$exception"
+      return exception.split("\\.".toRegex())
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt]: split with char

File: okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt (hunk 2 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt]: done!

File: okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt (hunk 3 above)
Comment (swankjesse) [reply okhttp/src/main/java/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt]: Fixed all!

**Pull request merged**